R/multiEquatError.r

```diff
@@ -1,19 +1,30 @@
-multiEquatError <- function (x1, x2, x3, difBound = 1, dependentDIF =FALSE, testletStr = NULL, verbose=TRUE ){
-       liste<- list(x1, x2, x3)
-       chk1 <- checkInput(liste)
-       if ( "derived.par" %in% colnames(x1) ) {                                 
-            obj  <- prepareAndCheckEatModelObject(liste, difBound=difBound, verbose=verbose)
-            link <- lapply(names(obj), FUN = function (dim ) { tripleEquatError(e1=obj[[dim]][[1]][,c("item", "est")], e2=obj[[dim]][[2]][,c("item", "est")], e3= obj[[dim]][[3]][,c("item", "est")], dependentDIF=dependentDIF, testletStr=testletStr, difBound=difBound, calledForEatModel = TRUE)})
-            names(link) <- names(obj)
-       } else {                                                                 
-            link <- tripleEquatError(e1=x1, e2=x2, e3=x3, dependentDIF=dependentDIF, testletStr=testletStr, difBound=difBound, calledForEatModel = FALSE)
-       }
-       return(link) }
 
+multiEquatError <- function (x1, x2, x3, difBound = 1, dependentDIF =FALSE, testletStr = NULL, verbose=TRUE){
+  lapply(list(x1, x2, x3), checkmate::assert_data_frame, min.cols = 2)
+  checkmate::assert_numeric(difBound, len = 1)
+  lapply(c(dependentDIF, verbose), checkmate::assert_logical, len = 1)
+  checkmate::assert_data_frame(testletStr, ncols = 2, col.names = "named", null.ok = TRUE)
+  #
+  if("derived.par" %in% colnames(x1)){
+    obj  <- prepareAndCheckEatModelObject(liste, difBound=difBound, verbose=verbose)
+    link <- lapply(names(obj), FUN = function (dim ) {
+      tripleEquatError(e1=obj[[dim]][[1]][,c("item", "est")],
+                       e2=obj[[dim]][[2]][,c("item", "est")],
+                       e3= obj[[dim]][[3]][,c("item", "est")],
+                       dependentDIF=dependentDIF, testletStr=testletStr, difBound=difBound, calledForEatModel = TRUE)})
+    names(link) <- names(obj)
+  } else {
+    link <- tripleEquatError(e1=x1, e2=x2, e3=x3, dependentDIF=dependentDIF,

```

Comment:
I think it would be better to pass the tested and possibly prepared objects `x1`, `x2`, `x3` to the function `tripleEquatError`, something like this:
```r
link <- tripleEquatError(e1=liste[[1]], e2=liste[[2]], e3=liste[[3]], 
    dependentDIF=dependentDIF, testletStr=testletStr, difBound=difBound, 
    calledForEatModel = FALSE)
```